sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/component.py

```diff
@@ -176,6 +177,21 @@ def _additional_includes(self):
     def _create_schema_for_validation(cls, context) -> Union[PathAwareSchema, Schema]:
         return InternalBaseComponentSchema(context=context)
 
+    def _validate(self, raise_error=False) -> MutableValidationResult:
+        if self._additional_includes is not None and self._additional_includes._validate().passed:
+            # update source path in case dependency file is in additional_includes
+            with self._resolve_local_code() as tmp_base_path:
+                origin_base_path, origin_source_path = self._base_path, self._source_path
+
+                try:
+                    self._base_path, self._source_path = \
```

Comment:
This is not safe right? Nodes created from one component func will share that component, and if we are going to make the create/validate parallel in the future for pipeline perf, here is something we need to aware.

Reply:
Agree that there can be perf issue. I keep the implementation to reuse the logic to resolve local code & we may cache validation result later to resolve perf issue.

But why do you think it's not safe?